presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraPartitionManager.java

```diff
@@ -114,35 +114,31 @@ public CassandraPartitionResult getPartitions(ConnectorTableHandle tableHandle,
             return ImmutableList.of();
         }
 
-        Set<List<Object>> partitionKeysSet = getPartitionKeysSet(table, tupleDomain);
+        List<Set<Object>> partitionKeysList = getPartitionKeysList(table, tupleDomain);
 
-        // empty filter means, all partitions
-        if (partitionKeysSet.isEmpty()) {
+        Set<List<Object>> filterList = Sets.cartesianProduct(partitionKeysList);
+        // empty filters means, all partitions
+        if (filterList.isEmpty()) {
```

Comment:
this `if` looks like not needed

Reply:
@kokosing notice that the comparison is done against `filterList` and not `partitionKeysList `. This is inline with the [original](https://github.com/prestodb/presto/blob/144438c58a24967146c5da97a59d0bf6fc255e6c/presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraPartitionManager.java#L120) implementation. `partitionKeysList` can be a list of [empty sets](https://github.com/aandis/presto/blob/c4b998c3e607c1c1d88809977114b6bf3e4cb4e3/presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraPartitionManager.java#L163) returned by [getPartitionKeysList](https://github.com/aandis/presto/blob/c4b998c3e607c1c1d88809977114b6bf3e4cb4e3/presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraPartitionManager.java#L128). So we need to take a cartesian product and check if the filters are actually empty.